src/memory/AmdFlash.hh

```diff
@@ -33,11 +33,17 @@ public:
 		bool oddZero : 1 = false;      // odd bytes are zero, not mirrored
 		size_t readMask = 0x03;        // read address mask
 	};
+	enum class DeviceInterface : uint16_t {
+		x8 = 0x0000,
+		x16 = 0x0001,
+		x8x16 = 0x0002,
+	};
```

Comment:
Is my understand of this enum correct?
* x8: chip has an 8-bit data bus (addresses are interpreted as 8-bit addresses)
* x16: chip has a 16-bit data bus.  (addresses are interpreted as 16-bit addresses)
* x8x16: chip can operate in either 8 or 16 bit mode.  And depending on the mode the address should be interpreted differently

While reading the code I was expecting to see special cases for x16, but I saw none. Later I realized that's maybe because there currently are no chips yet that specify x16 (and maybe because of that handling of this case is missing?)

Do chips with x16 make sense in the context of MSX (an 8-bit system).  Maybe, but then you loose half the capacity of the ROM?

Reply:
At some point I was considering adding `read16()`, `peek16()`, `write16()` and similar methods. Indeed x16 chips could be used on MSX if you add a multiplexer between the flash ROM and the bus, controlled by A0. The x8x16 chips also internally are more x16 than they are x8, so implementation-wise it could potentially make the code a bit more internally consistent / logical.

However since there is no current hardware using x16 chips, I decided to not add extra complexity for that at the moment. This x16 entry here is a leftover of that consideration, I’ll remove it.